cpp/src/arrow/json/parser.cc

```diff
@@ -504,6 +514,9 @@ class RawBuilderSet {
         }
         return Status::OK();
       }
+      case Kind::kNumberOrString: {
+        return Cast<Kind::kNumberOrString>(builder)->AppendNull();
```

Comment:
Same here: move this up below the `kString` case?

Reply:
done